runtimevar/blobvar/blobvar_test.go

```diff
@@ -88,3 +91,66 @@ func (verifyAs) ErrorCheck(v *runtimevar.Variable, err error) error {
 	}
 	return nil
 }
+
+func TestOpenVariable(t *testing.T) {
+	dir, err := ioutil.TempDir("", "gcdk-blob-var-example")
+	if err != nil {
+		t.Fatal(err)
+	}
+	if err := ioutil.WriteFile(filepath.Join(dir, "myvar.json"), []byte(`{"Foo": "Bar"}`), 0666); err != nil {
+		t.Fatal(err)
+	}
+	if err := ioutil.WriteFile(filepath.Join(dir, "myvar.txt"), []byte("hello world!"), 0666); err != nil {
+		t.Fatal(err)
+	}
+	defer os.RemoveAll(dir)
+
+	tests := []struct {
+		URL          string
+		WantErr      bool
+		WantWatchErr bool
+		Want         interface{}
+	}{
+		// Variable construction succeeds, but myvar does not exist.
+		{"blob://myvar?bucket=mem://&decoder=string", false, true, nil},
+		// Variable construction fails because the directory dirnotfound does not
+		// exist, so the Bucket creation fails.
+		{"blob://myvar.txt?bucket=file:///dirnotfound&decoder=string", true, false, nil},
+		// Variable construction succeeds, but filenotfound does not exist so Watch
+		// returns an error.
+		{"blob://filenotfound?bucket=file://" + dir + "&decoder=string", false, true, nil},
+		// Variable construction fails due to missing bucket arg.
+		{"blob://myvar.txt?decoder=string", true, false, nil},
+		// Variable construction fails due to invalid wait arg.
+		{"blob://myvar.txt?bucket=file://" + dir + "&decoder=string&wait=notaduration", true, false, nil},
+		// Variable construction fails due to invalid decoder arg.
+		{"blob://myvar.txt?bucket=file://" + dir + "&decoder=notadecoder", true, false, nil},
+		// Variable construction fails due to invalid arg.
+		{"blob://myvar.txt?bucket=file://" + dir + "&decoder=string&param=value", true, false, nil},
+		// Working example with string decoder and wait.
```

Comment:
Consider add an example with no decoder parameter so that it uses default bytes decoder.

Reply:
Good idea, done.